client/state/query-client.ts

```diff
@@ -1,10 +1,22 @@
 import { throttle } from 'lodash';
-import { QueryClient } from 'react-query';
+import { QueryClient, QueryCache } from 'react-query';
 import { persistQueryClient } from 'react-query/persistQueryClient-experimental';
+import { BASE_STALE_TIME } from 'calypso/data/marketplace/constants';
 import { shouldPersist, MAX_AGE, SERIALIZE_THROTTLE } from 'calypso/state/initial-state';
 import { getPersistedStateItem, storePersistedStateItem } from 'calypso/state/persisted-state';
 import { shouldDehydrateQuery } from './should-dehydrate-query';
 
+const sharedCache = new QueryCache();
+
+export function createServerQueryClient() {
+	const queryClient = new QueryClient( {
+		defaultOptions: { queries: { cacheTime: MAX_AGE, staleTime: BASE_STALE_TIME } },
+		queryCache: sharedCache,
+	} );
+
+	return queryClient;
+}
```

Comment:
I would create a separate module for this, like `query-client-ssr.ts`. The `client/state/query-client.ts` one imports many things that are irrelevant on server, like code for serializing Redux state to `window.indexedDB` etc.

Reply:
Sounds good!
I Moved the query-client for SSR to a dedicated file as suggested.